src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskControllerTest.java

```diff
@@ -110,6 +114,41 @@ public void should_return_a_204_when_claiming_a_task_by_id() {
             .statusCode(HttpStatus.NO_CONTENT.value());
     }
 
+    @Test
+    public void should_return_a_200_with_search_results() {
+
+        String ccdId1 = ccdIdGenerator.generate();
+        String ccdId2 = ccdIdGenerator.generate();
+
+        given
+            .iCreateATaskWithCcdId(ccdId1)
+            .and()
+            .iCreateATaskWithCcdId(ccdId2);
+
+        SearchParameters searchParameters = new SearchParameters(
+            emptyList(),
+            emptyList(),
+            emptyList(),
+            emptyList(),
+            ccdId2,
+            null,
+            null,
+            null
+        );
+        Response result = given()
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .headers(authorizationHeadersProvider.getLawFirmAAuthorization())
+            .body(new SearchTaskRequest(singletonList(searchParameters)))
+            .when()
+            .post("task");
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("tasks.size()", equalTo(1))
+            .body("tasks[0].caseData.reference", equalTo(ccdId2))
+        ;
```

Comment:
could we include functional tests to cover the other 2 AC's; i.e.
1. Search by jurisdiction(s), locationId(s) & state = unassigned and verify its for a list of the active and unassigned tasks
2. Search by jurisdiction(s), locationId(s) and verify its for a list of the active tasks (assigned and unassigned) for that Location (or Locations)

Reply:
Thanks, Ben I've added extra tests to satisfy these ACs.